tests/components/juicenet/test_config_flow.py

```diff
@@ -0,0 +1,123 @@
+"""Test the JuiceNet config flow."""
+import aiohttp
+from asynctest import patch
+from asynctest.mock import MagicMock
+from pyjuicenet import TokenError
+
+from homeassistant import config_entries, setup
+from homeassistant.components.juicenet.const import DOMAIN
+from homeassistant.const import CONF_ACCESS_TOKEN
+
+
+def _mock_juicenet_return_value(get_devices=None):
+    juicenet_mock = MagicMock()
+    type(juicenet_mock).get_devices = MagicMock(return_value=get_devices)
+    return juicenet_mock
+
+
+async def test_form(hass):
+    """Test we get the form."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == "form"
+    assert result["errors"] == {}
+
+    with patch(
+        "homeassistant.components.juicenet.config_flow.Api.get_devices",
+        return_value=MagicMock(),
+    ), patch(
+        "homeassistant.components.juicenet.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "homeassistant.components.juicenet.async_setup_entry", return_value=True
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {CONF_ACCESS_TOKEN: "access_token"}
+        )
+
+    assert result2["type"] == "create_entry"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5491418)